### PR TITLE
New task: Configure a central syslog server

### PR DESCRIPTION
**Pull request by cwickert**
Feel free to make changes as you like. Please take care of merging this while I'm away over the weekend. TIA!
[task-configure-central-syslog-server_color_en.pdf](https://github.com/SUSE/doc-modular/files/7170776/task-configure-central-syslog-server_color_en.pdf)

### CODE REVIEW

File: xml/task-configure-syslog-server.xml [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+      </callout>
+      <callout arearefs="co-tuning-syslog-server-port">
+       <para>
+        Port for <systemitem class="daemon">rsyslogd</systemitem> to listen on.
+        Select a privileged port below 1024. The default is 514.
+       </para>
+      </callout>
+     </calloutlist>
+     <important>
+      <title>TCP vs. UDP protocol</title>
```
Comment (lproven) [inline xml/task-configure-syslog-server.xml:156]: How about the full word, "versus", instead of the abbreviation?

File: xml/task-configure-syslog-server.xml [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+    server. Repeat this procedure for all machines that should log remotely.
+   </para>
+  </section>
+ </section>
+
+ <section xml:id="summary-configure-central-syslog-server">
+  <title>Summary</title>
+  <para>
+   You have now configured one or more hosts for remote logging to your central
+   log server. This allows you to get a quick to get an overview of events in
```
Comment (lproven) [inline xml/task-configure-syslog-server.xml:274]:
```suggestion
   syslog server. This allows you to get a quick overview of events
```

File: xml/task-configure-syslog-server.xml [hunk 3]
```diff
@@ -0,0 +1,355 @@
...
+   in your network.
```
Comment (lproven) [inline xml/task-configure-syslog-server.xml:275]: "On your network"?

File: xml/task-configure-syslog-server.xml [hunk 4]
```diff
@@ -0,0 +1,355 @@
...
+    </listitem>
+   </varlistentry>
+  </variablelist>
+ </section>
+
+ <section xml:id="next-configure-central-syslog-server">
+  <title>Next steps</title>
+  <para>
+   This basic setup does not include encryption and is only suitable for
+   trusted internal networks. TLS encryption is strongly recommended but
```
Comment (lproven) [inline xml/task-configure-syslog-server.xml:318]:
```suggestion
   trusted internal networks. TLS encryption is strongly recommended, but
```

File: xml/task-configure-syslog-server.xml [hunk 5]
```diff
@@ -0,0 +1,355 @@
...
+   requires a certificate infrastructure to be set up first.
```
Comment (lproven) [inline xml/task-configure-syslog-server.xml:319]:
```suggestion
   requires you to set up a certificate infrastructure first.
```

Comment (lproven) [reply xml/task-configure-syslog-server.xml:319]: Rewording to avoid passive voice.

File: xml/task-configure-syslog-server.xml [hunk 6]
```diff
@@ -0,0 +1,355 @@
...
+  </para>
+  <para>
+   In this setup all messages from remote hosts will be treated the same on the
```
Comment (lproven) [inline xml/task-configure-syslog-server.xml:322]:
```suggestion
   In this configuration, all messages from remote hosts will be treated the same on the
```

File: xml/task-configure-syslog-server.xml [hunk 7]
```diff
@@ -0,0 +1,355 @@
...
+   central syslog server. Consider filtering messages into separate files by
+   remote host or classify them by message category.
```
Comment (lproven) [inline xml/task-configure-syslog-server.xml:324]:
```suggestion
   central syslog server. Consider filtering messages into separate files
   according to the remote host, or classifying them by message category.
```

File: xml/task-configure-syslog-server.xml [hunk 8]
```diff
@@ -0,0 +1,355 @@
...
+  </listitem>
+ </itemizedlist>
+</section>
+
+ <section xml:id="introduction-configure-central-syslog-server">
+  <title>Introduction</title>
+  <para>
+   System log data can be forwarded from individual systems to a central
+   syslog server on the network. This allows administrators to get an overview
+   of events on all hosts and prevents attackers that succeed in taking over a
```
Comment (lproven) [inline xml/task-configure-syslog-server.xml:69]:
```suggestion
   of events on all hosts, and prevents attackers that succeed in taking over a
```

File: xml/task-configure-syslog-server.xml [hunk 9]
```diff
@@ -0,0 +1,355 @@
...
+   system from manipulating system logs to cover their tracks.
```
Comment (lproven) [inline xml/task-configure-syslog-server.xml:70]:
```suggestion
   system from manipulating the system logs to cover their tracks.
```

File: xml/task-configure-syslog-server.xml [hunk 10]
```diff
@@ -0,0 +1,355 @@
...
+   <title>Setting up the central syslog server</title>
+   <!--
+    <para>
+    This section describes a basic syslog forwarding setup on &sle;.
+    </para>
+   -->
+   <procedure xml:id="pro-configure-central-syslog-server">
+    <title>Configure the central <systemitem>rsyslog</systemitem> server</title>
+    <para>
+     To set up a central log server, perform the following steps:
```
Comment (lproven) [inline xml/task-configure-syslog-server.xml:113]:
```suggestion
     To set up a central syslog server, perform the following steps:
```

File: xml/task-configure-syslog-server.xml [hunk 11]
```diff
@@ -0,0 +1,355 @@
...
+        Port for <systemitem class="daemon">rsyslogd</systemitem> to listen on.
+        Select a privileged port below 1024. The default is 514.
+       </para>
+      </callout>
+     </calloutlist>
+     <important>
+      <title>TCP vs. UDP protocol</title>
+      <para>
+       Traditionally syslog uses the UDP protocol to transmit log messages over
+       the network. This involves less overhead but lacks reliability. Log
```
Comment (lproven) [inline xml/task-configure-syslog-server.xml:159]:
```suggestion
       the network. This involves less overhead, but lacks reliability. Log
```

File: xml/task-configure-syslog-server.xml [hunk 12]
```diff
@@ -0,0 +1,355 @@
...
+    <step>
+     <para>Open the respective port in the firewall. For <systemitem
+      class="daemon">firewalld</systemitem> with TCP on port 514 run:
+     </para>
+<screen>&prompt.sudo;<command>firewall-cmd --add-port <replaceable>514/tcp</replaceable> --permanent</command>
+&prompt.sudo;<command>firewall-cmd --reload</command></screen>
+    </step>
+   </procedure>
+   <para>
+    You have now configured the central logging server. Next, configure the
```
Comment (lproven) [inline xml/task-configure-syslog-server.xml:204]:
```suggestion
    You have now configured the central syslog server. Next, configure the
```

File: xml/task-configure-syslog-server.xml [hunk 13]
```diff
@@ -0,0 +1,355 @@
...
+    clients for remote logging.
+   </para>
+  </section>
+  <section >
+   <title>Set up the client machines</title>
+   <procedure xml:id="pro-configure-syslog-client">
+    <title>Configure a <guimenu>rsyslog</guimenu> instance for remote logging</title>
+    <para>
+     To configure a machine for remote logging on a central log server, perform
```
Comment (lproven) [inline xml/task-configure-syslog-server.xml:213]:
```suggestion
     To configure a machine for remote logging on a central syslog server, perform
```

File: xml/task-configure-syslog-server.xml [hunk 14]
```diff
@@ -0,0 +1,355 @@
...
+    <step>
+     <para>
+      Edit the configuration file
+      <filename>/etc/rsyslog.d/remote.conf</filename>.
+     </para>
+    </step>
+    <step>
+     <para>
+      Uncomment the appropriate line (TCP or UDP) and replace
+      <literal>remote-host</literal> with the address of the central log server
```
Comment (lproven) [inline xml/task-configure-syslog-server.xml:225]:
```suggestion
      <literal>remote-host</literal> with the address of the central syslog server
```

File: xml/task-configure-syslog-server.xml [hunk 15]
```diff
@@ -0,0 +1,355 @@
...
+     </para>
+     <screen>&prompt.user;<command>logger "hello world"</command></screen>
+     <para>
+      The log message <literal>hello world</literal> should now appear on the
+      central syslog server.
+     </para>
+    </step>
+   </procedure>
+   <para>
+    You have now configured a machine for remote logging to your central log
```
Comment (lproven) [inline xml/task-configure-syslog-server.xml:264]:
```suggestion
    You have now configured a machine for remote logging to your central syslog
```

File: xml/task-configure-syslog-server.xml [hunk 16]
```diff
@@ -0,0 +1,355 @@
...
+   You have now configured one or more hosts for remote logging to your central
+   log server. This allows you to get a quick to get an overview of events in
+   in your network.
+  </para>
+ </section>
+ 
+ <section xml:id="troubleshooting-configure-central-syslog-server">
+  <title>Troubleshooting</title>
+  <para>
+   In case the test log message does not appear on the log server, perform the
```
Comment (lproven) [inline xml/task-configure-syslog-server.xml:282]:
```suggestion
   In case the test log message does not appear on the syslog server, perform the
```

File: xml/task-configure-syslog-server.xml [hunk 17]
```diff
@@ -0,0 +1,355 @@
...
+     <para>
+      If you made an error in the configuration of <systemitem
+       class="daemon">rsyslog</systemitem>, the daemon might refuse to start.
+      Check it is running with
+      <command>systemctl status rsyslog.service</command>. If the
+      service is down, the output includes additional information about the
+      reason.
+     </para>
+     <para>
+      Run this check on both the log server and the remote logging clients.
```
Comment (lproven) [inline xml/task-configure-syslog-server.xml:298]:
```suggestion
      Run this check on both the syslog server and the remote logging clients.
```

File: xml/task-configure-syslog-server.xml [hunk 18]
```diff
@@ -0,0 +1,355 @@
...
+     </para>
+    </listitem>
+   </varlistentry>
+   <varlistentry>
+    <term>Is the firewall open?</term>
+    <listitem>
+     <para>
+      Check if the firewall on the log server is open with
```
Comment (lproven) [inline xml/task-configure-syslog-server.xml:306]:
```suggestion
      Check if the firewall on the syslog server is open with
```

**Review by lproven — CHANGES_REQUESTED** [review]

File: xml/task-configure-syslog-server.xml (hunk 7 above)
Comment (cwickert) [reply xml/task-configure-syslog-server.xml:324]: This strikes me as odd. "filter by" is common, I never heard "filter" in combination with "according to".

File: xml/task-configure-syslog-server.xml (hunk 10 above)
Comment (cwickert) [reply xml/task-configure-syslog-server.xml:113]: I'm trying to avoid duplication, so I usually use "syslog server" in the first occurrence in a section/paragraph and "log server" afterwards.

File: xml/task-configure-syslog-server.xml (hunk 5 above)
Comment (dmpop) [reply xml/task-configure-syslog-server.xml:319]: For some reason I was notified of this comment, so I feel the urge to chime in. :smile:  Liam's suggestion sounds much better. The passive voice in general, and in this case in particular, sounds contrived. And _you_ doesn't refer to a specific person but a collective you.

File: xml/task-configure-syslog-server.xml (hunk 3 above)
Comment (cwickert) [reply xml/task-configure-syslog-server.xml:275]: Yes, changed, thanks.

File: xml/task-configure-syslog-server.xml (hunk 17 above)
Comment (cwickert) [reply xml/task-configure-syslog-server.xml:306]: [a] It's not repetition but a mix of the long and the short phrase to _avoid_ repetition. Just like you use an article instead of a noun when it's clear what you are referring to. We do this all the time, e.g. we first use 'DNS server' and then 'the server' or 'file permissions' and then just 'permissions'. This eases reading and makes the text shorter.
[b] It is consistent: I would argue that the text is consistent. For the first occurrence in a section I use the long term, then the shorter one. 

IHMO readability is just as important as consistency, and in this case wins because there is no ambiguity. Do you see any room for misinterpretations? What other server could this sentence refer to?

Comment (sknorr) [reply xml/task-configure-syslog-server.xml:306]: >[a] in docs, repetition is so bad as in normal prose

I concur with Liam here -- exact repetition helps with terminology and translation.

Comment (lproven) [reply xml/task-configure-syslog-server.xml:306]: Thanks.

We are basically only and always talking about servers. This logging server collects logs from other servers and nothing but other servers. So I think we need to be specific.

Comment (cwickert) [reply xml/task-configure-syslog-server.xml:306]: > We are basically only and always talking about servers. This logging server collects logs from other servers and nothing but other servers.

No, the machines that log remotely are not necessarily servers and we explicitly call them clients here. Whether or not they offer any service over the network is a completely different question, that is not been dealt with here.

Anyway, I give in and will change it accordingly.

File: xml/task-configure-syslog-server.xml (hunk 5 above)
Comment (cwickert) [reply xml/task-configure-syslog-server.xml:327]: To avoid this discussion, I simplified
`requires a certificate infrastructure to be set up first.` to
`requires a certificate infrastructure.`
When or who sets it up is a different question, we just say it's a requirement.

**Comment by cwickert** [comment]
I have now replaced all but one instance of "log server" with "syslog server". The only remaining occurrence of "log server" is in line 109, where we have "syslog server" right before it in line 108 repeating this really sounds awkward.

@sknorr or @dmpop, please be so kind as to review and approve this.

File: xml/task-configure-syslog-server.xml (hunk 5 above)
Comment (cwickert) [reply xml/task-configure-syslog-server.xml]: > For some reason I was notified of this comment,
@dmpop, you were notified because I requested a review from you.

File: xml/task-configure-syslog-server.xml [hunk 19]
```diff
@@ -0,0 +1,364 @@
...
+     linkend="FIXME"/> for configuration instructions.
+    </para>
+   </listitem>
+   -->
+   <listitem>
+    <para>
+     The <package>rsyslog</package> package is installed on all machines.
+     If not, run <command>zypper in yast2-mail</command> to install it.
+    </para>
+    <!-- <screen>&prompt.root;<command>zypper in rsyslog</command></screen> -->
```
Comment (sknorr) [inline xml/task-configure-syslog-server.xml:100]: I am confused by these installation instructions. My own system with Leap has `yast2-mail` installed, but the `rsyslog` package is missing. The paragraph also does a bad job at explaining cause & effect here -- is rsyslog a package required by YaST somehow?

```diff
> zypper se yast2-mail

S  | Name       | Summary                    | Type
 --+------------+----------------------------+-----------
i+ | yast2-mail | YaST2 - Mail Configuration | package
   | yast2-mail | YaST2 - Mail Configuration | srcpackage


> zypper se rsyslog

S | Name                         | Summary                          | Type
 -+------------------------------+----------------------------------+-----------
  | pcp-pmda-rsyslog             | Performance Co-Pilot (PCP) met-> | package
  | rsyslog                      | The enhanced syslogd for Linux-> | package
  | rsyslog                      | The enhanced syslogd for Linux-> | srcpackage
[...]
```

File: xml/task-configure-syslog-server.xml [hunk 20]
```diff
@@ -0,0 +1,364 @@
...
+   <!--
+    <para>
+    This section describes a basic syslog forwarding setup on &sle;.
+    </para>
+   -->
+   <procedure xml:id="pro-configure-central-syslog-server">
+    <title>Configure the central <systemitem>rsyslog</systemitem> server</title>
+    <para>
+     To set up a central syslog server, perform the following steps:
+    </para>
```
Comment (sknorr) [inline xml/task-configure-syslog-server.xml:122]: Why create a procedure with title and a preamble in a section that contains nothing but this one procedure anyway? Suggestion: delete the procedure's title tag, move the preamble para above the procedure, and integrate or delete the commented para that already exists in between section title and begin of the procedure.
(And maybe remove the xml:id from the procedure.)

File: xml/task-configure-syslog-server.xml [hunk 21]
```diff
@@ -0,0 +1,364 @@
...
+    <step>
+     <para>
+      Edit the configuration file
```
Comment (sknorr) [inline xml/task-configure-syslog-server.xml:125]: This step does not tell you anything about editing things, soo... maybe reword like this:

```suggestion
      In an editor, open the configuration file
```

File: xml/task-configure-syslog-server.xml [hunk 22]
```diff
@@ -0,0 +1,364 @@
...
+       <para>
+        IP address of the interface for <systemitem
+         class="daemon">rsyslogd</systemitem> to listen on. If no address is
+        given, the daemon listens on all interfaces.
+       </para>
+      </callout>
+      <callout arearefs="co-tuning-syslog-server-port">
+       <para>
+        Port for <systemitem class="daemon">rsyslogd</systemitem> to listen on.
+        Select a privileged port below 1024. The default is 514.
```
Comment (sknorr) [inline xml/task-configure-syslog-server.xml:159]: I am a moderate fan of putting port numbers into literals. I won't be offended if you disagree though.

```suggestion
        Select a privileged port below <literal>1024</literal>. The default port is
        <literal>514</literal>.
```

File: xml/task-configure-syslog-server.xml [hunk 23]
```diff
@@ -0,0 +1,364 @@
...
+       </para>
+      </callout>
+     </calloutlist>
+     <important>
+      <title>TCP versus UDP protocol</title>
+      <para>
+       Traditionally syslog uses the UDP protocol to transmit log messages over
+       the network. This involves less overhead, but lacks reliability. Log
```
Comment (sknorr) [inline xml/task-configure-syslog-server.xml:167]: Improve wording:

```suggestion
       the network. This creates less overhead, but is less reliable. Log
```

File: xml/task-configure-syslog-server.xml [hunk 24]
```diff
@@ -0,0 +1,364 @@
...
+       messages can get lost under high load.
+       <!-- cwickert 2021-03-02 Original text before shortening -->
+       <!-- The TCP protocol is more
+       reliable. Messages will only get lost under
+       <emphasis>constant</emphasis> high load, which should not occur under
+       normal circumstances.
+      </para>
+      <para>
+       Since the advantages of centralized logging could suffer from
+       unreliability, using TCP recommended. -->
```
Comment (sknorr) [inline xml/task-configure-syslog-server.xml:177]: Not a huge fan of keeping every possible prior wording in the file itself. Most likely, three years down the road, someone will just cut it anyway.

```suggestion
```

File: xml/task-configure-syslog-server.xml [hunk 25]
```diff
@@ -0,0 +1,364 @@
...
+        Select a privileged port below 1024. The default is 514.
+       </para>
+      </callout>
+     </calloutlist>
+     <important>
+      <title>TCP versus UDP protocol</title>
+      <para>
+       Traditionally syslog uses the UDP protocol to transmit log messages over
+       the network. This involves less overhead, but lacks reliability. Log
+       messages can get lost under high load.
```
Comment (sknorr) [inline xml/task-configure-syslog-server.xml:168]: If the old wording below is actually more correct (as is suggested), the new wording should continue to include "constant" imo, as that seems important.

```suggestion
       messages can get lost under constant high load.
```

File: xml/task-configure-syslog-server.xml [hunk 26]
```diff
@@ -0,0 +1,364 @@
...
+      </para>
+      <para>
+       The TCP protocol is more reliable and should be preferred over UDP.
+      </para>
+     </important>
+     <note>
+      <title><literal>UDPServerAddress</literal> with TCP</title>
+      <para>
+       The <literal>$UDPServerAddress</literal> configuration parameter in the
+       TCP example is no error. Despite its name it is used for both TCP and
```
Comment (sknorr) [inline xml/task-configure-syslog-server.xml:187]: I think this works better; and the comma is rather conventional in our docs:

```suggestion
       TCP example is not an error. Despite its name, it is used for both TCP and
```

File: xml/task-configure-syslog-server.xml [hunk 27]
```diff
@@ -0,0 +1,364 @@
...
+    <step>
+     <para>
+      Save the file.
+     </para>
+    </step>
+    <step>
+     <para>
+      Restart the <systemitem class="daemon">rsyslog</systemitem> service:
+     </para>
+<screen>&prompt.sudo;<command>systemctl restart rsyslog.service</command></screen>
```
Comment (sknorr) [inline xml/task-configure-syslog-server.xml:201]: As far as I know, it was never started, it was only installed. Does running `systemctl restart` for a service that is not yet running work as expected?

File: xml/task-configure-syslog-server.xml [hunk 28]
```diff
@@ -0,0 +1,364 @@
...
+    </step>
+    <step>
+     <para>Open the respective port in the firewall. For <systemitem
+      class="daemon">firewalld</systemitem> with TCP on port 514 run:
```
Comment (sknorr) [inline xml/task-configure-syslog-server.xml:205]: Comma me:

```suggestion
      class="daemon">firewalld</systemitem> with TCP on port 514, run:
```

File: xml/task-configure-syslog-server.xml [hunk 29]
```diff
@@ -0,0 +1,364 @@
...
+   </para>
+  </section>
+  <section >
+   <title>Set up the client machines</title>
+   <procedure xml:id="pro-configure-syslog-client">
+    <title>Configure a <guimenu>rsyslog</guimenu> instance for remote logging</title>
+    <para>
+     To configure a system for remote logging on a central syslog server,
+     perform the following steps:
+    </para>
```
Comment (sknorr) [inline xml/task-configure-syslog-server.xml:223]: Same comment as before -- this is another section that only has a single procedure in it, it would make more sense to delete the procedure title and move the procedure preamble up.

File: xml/task-configure-syslog-server.xml [hunk 30]
```diff
@@ -0,0 +1,364 @@
...
+    <step>
+     <para>
+      Edit the configuration file
```
Comment (sknorr) [inline xml/task-configure-syslog-server.xml:226]: Same as above.

```suggestion
      In an editor, open the configuration file
```

File: xml/task-configure-syslog-server.xml [hunk 31]
```diff
@@ -0,0 +1,364 @@
...
+    </step>
+    <step>
+     <para>
+      Restart the <systemitem class="daemon">rsyslog</systemitem> service:
+     </para>
+     <screen>&prompt.sudo;<command>systemctl restart rsyslog.service</command></screen>
+    </step>
+    <step>
+     <para>
+      Verify the proper function of the syslog forwarding:
```
Comment (sknorr) [inline xml/task-configure-syslog-server.xml:263]: Not much shorter but imo easier to understand:

```suggestion
      Make sure that syslog forwarding works:
```

File: xml/task-configure-syslog-server.xml [hunk 32]
```diff
@@ -0,0 +1,364 @@
...
+<screen>&prompt.sudo;<command>firewall-cmd --add-port <replaceable>514/tcp</replaceable> --permanent</command>
+&prompt.sudo;<command>firewall-cmd --reload</command></screen>
+    </step>
+   </procedure>
+   <para>
+    You have now configured the central syslog server. Next, configure clients
+    for remote logging.
+   </para>
+  </section>
+  <section >
```
Comment (sknorr) [inline xml/task-configure-syslog-server.xml:216]: This section has no xml:id.

File: xml/task-configure-syslog-server.xml [hunk 33]
```diff
@@ -0,0 +1,364 @@
...
+   <title>Set up the client machines</title>
```
Comment (sknorr) [inline xml/task-configure-syslog-server.xml:217]: Gerund:

```suggestion
   <title>Setting up the client machines</title>
```

File: xml/task-configure-syslog-server.xml [hunk 34]
```diff
@@ -0,0 +1,364 @@
...
+    <step>
+     <para>
+      Save the file.
+     </para>
+    </step>
+    <step>
+     <para>
+      Restart the <systemitem class="daemon">rsyslog</systemitem> service:
+     </para>
+     <screen>&prompt.sudo;<command>systemctl restart rsyslog.service</command></screen>
```
Comment (sknorr) [inline xml/task-configure-syslog-server.xml:259]: The issue of restarting a service that may not be running applies here as well.

File: xml/task-configure-syslog-server.xml [hunk 35]
```diff
@@ -0,0 +1,364 @@
...
+     <screen>&prompt.user;<command>logger "hello world"</command></screen>
+     <para>
+      The log message <literal>hello world</literal> should now appear on the
+      central syslog server.
+     </para>
+    </step>
+   </procedure>
+   <para>
+    You have now configured a system for remote logging to your central syslog
+    server. Repeat this procedure for all systems that should log remotely.
```
Comment (sknorr) [inline xml/task-configure-syslog-server.xml:274]: "for" -> "on"? To me, it seems like it would make sense (because you're changing a config file on that machine).

```suggestion
    server. Repeat this procedure on all systems that should log remotely.
```

File: xml/task-configure-syslog-server.xml [hunk 36]
```diff
@@ -0,0 +1,364 @@
...
+   </para>
+  </section>
+ </section>
+
+ <section xml:id="summary-configure-central-syslog-server">
+  <title>Summary</title>
+  <para>
+   You have configured one or more hosts for remote logging to your central
+   syslog server. This allows you to get a quick overview of events on your
+   network.
```
Comment (sknorr) [inline xml/task-configure-syslog-server.xml:284]: Sounds better to me, but not a native speaker I am:

```suggestion
   syslog server. This allows you to get a quick overview of events
   within your network.
```

File: xml/task-configure-syslog-server.xml [hunk 37]
```diff
@@ -0,0 +1,364 @@
...
+  </para>
+ </section>
+ 
+ <section xml:id="troubleshooting-configure-central-syslog-server">
+  <title>Troubleshooting</title>
+  <para>
+   In case the test log message does not appear on the syslog server, perform
```
Comment (sknorr) [inline xml/task-configure-syslog-server.xml:291]: Wordy:

```suggestion
   If the test log message does not appear on the syslog server, perform
```

File: xml/task-configure-syslog-server.xml [hunk 38]
```diff
@@ -0,0 +1,364 @@
...
+    <!-- <screen>&prompt.root;<command>zypper in rsyslog</command></screen> -->
+   </listitem>
+  </itemizedlist>
+ </section>
+
+ <section xml:id="configure-configure-central-syslog-server">
+  <title>Setting up the central syslog server</title>
+  <para>
+   Setting up a central syslog server consists of two parts. First you configure
+   the central log server, then the clients for remote logging.
```
Comment (sknorr) [inline xml/task-configure-syslog-server.xml:109]: An (imo) more accurate way to describe the task that also avoids the awkwardness you mentioned about repeating the word "syslog server":

```suggestion
   Setting up central logging with <systemitem class="daemon">rsyslog</systemitem>
   consists of two parts: Configuring a central syslog server and configuring
   clients to log remotely.
```

File: xml/task-configure-syslog-server.xml (hunk 18 above)
Comment (sknorr) [reply xml/task-configure-syslog-server.xml:315]: I would support Liam's suggestion (unsurprising, I guess ...)

File: xml/task-configure-syslog-server.xml [hunk 39]
```diff
@@ -0,0 +1,364 @@
...
+  <title>Next steps</title>
+  <para>
+   This basic setup does not include encryption and is only suitable for
+   trusted internal networks. TLS encryption is strongly recommended, but
+   requires a certificate infrastructure.
+  </para>
+  <para>
+   In this configuration, all messages from remote hosts will be treated the
+   same on the central syslog server. Consider filtering messages into separate
+   files by remote host or classify them by message category.
```
Comment (sknorr) [inline xml/task-configure-syslog-server.xml:333]: parallel phrasing (filter_ing_ + classify_ing_)

```suggestion
   files by remote host or classifying them by message category.
```

**Review by sknorr — COMMENTED** [review]
I am _sorry._ But this is probably not what you wanted. I tried to make suggestions though, so I hope integrating my feedback is painless irregardful.

File: xml/task-configure-syslog-server.xml [hunk 40]
```diff
@@ -0,0 +1,364 @@
...
+   system from manipulating system logs to cover their tracks.
+  </para>
+ </section>
+
+ <section xml:id="requirements-configure-central-syslog-server">
+  <title>Requirements</title>
+  <itemizedlist>
+   <listitem>
+    <para>
+     You have installed your product and your system is up and running.
```
Comment (dmpop) [inline xml/task-configure-syslog-server.xml:79]: What product? What system? It's the first time you mention them.

File: xml/task-configure-syslog-server.xml [hunk 41]
```diff
@@ -0,0 +1,364 @@
...
+    <para>
+     You have set up <literal>NTP</literal> on all machines. Refer to <xref
+     linkend="FIXME"/> for configuration instructions.
+    </para>
+   </listitem>
+   -->
+   <listitem>
+    <para>
+     The <package>rsyslog</package> package is installed on all machines.
+     If not, run <command>zypper in yast2-mail</command> to install it.
```
Comment (dmpop) [inline xml/task-configure-syslog-server.xml:98]:
```suggestion
     Otherwise, run <command>zypper in yast2-mail</command> to install it.
```

File: xml/task-configure-syslog-server.xml [hunk 42]
```diff
@@ -0,0 +1,364 @@
...
+    </para>
+    <!-- <screen>&prompt.root;<command>zypper in rsyslog</command></screen> -->
+   </listitem>
+  </itemizedlist>
+ </section>
+
+ <section xml:id="configure-configure-central-syslog-server">
+  <title>Setting up the central syslog server</title>
+  <para>
+   Setting up a central syslog server consists of two parts. First you configure
```
Comment (dmpop) [inline xml/task-configure-syslog-server.xml:108]:
```suggestion
   Setting up a central syslog server consists of two parts. First, configure
```

File: xml/task-configure-syslog-server.xml [hunk 43]
```diff
@@ -0,0 +1,364 @@
...
+       <para>
+        Port for <systemitem class="daemon">rsyslogd</systemitem> to listen on.
+        Select a privileged port below 1024. The default is 514.
+       </para>
+      </callout>
+     </calloutlist>
+     <important>
+      <title>TCP versus UDP protocol</title>
+      <para>
+       Traditionally syslog uses the UDP protocol to transmit log messages over
```
Comment (dmpop) [inline xml/task-configure-syslog-server.xml:166]: Traditionally? Not the word you'd use in a tech doc. Perhaps by default?

File: xml/task-configure-syslog-server.xml [hunk 44]
```diff
@@ -0,0 +1,364 @@
...
+       reliable. Messages will only get lost under
+       <emphasis>constant</emphasis> high load, which should not occur under
+       normal circumstances.
+      </para>
+      <para>
+       Since the advantages of centralized logging could suffer from
+       unreliability, using TCP recommended. -->
+      </para>
+      <para>
+       The TCP protocol is more reliable and should be preferred over UDP.
```
Comment (dmpop) [inline xml/task-configure-syslog-server.xml:180]:
```suggestion
       The TCP protocol is more reliable and it should be used instead of UDP.
```

File: xml/task-configure-syslog-server.xml [hunk 45]
```diff
@@ -0,0 +1,364 @@
...
+     </para>
+    </step>
+    <step>
+     <para>
+      Restart the <systemitem class="daemon">rsyslog</systemitem> service:
+     </para>
+<screen>&prompt.sudo;<command>systemctl restart rsyslog.service</command></screen>
+    </step>
+    <step>
+     <para>Open the respective port in the firewall. For <systemitem
```
Comment (dmpop) [inline xml/task-configure-syslog-server.xml:204]: Respective to what?

File: xml/task-configure-syslog-server.xml [hunk 46]
```diff
@@ -0,0 +1,364 @@
...
+    for remote logging.
+   </para>
+  </section>
+  <section >
+   <title>Set up the client machines</title>
+   <procedure xml:id="pro-configure-syslog-client">
+    <title>Configure a <guimenu>rsyslog</guimenu> instance for remote logging</title>
+    <para>
+     To configure a system for remote logging on a central syslog server,
+     perform the following steps:
```
Comment (dmpop) [inline xml/task-configure-syslog-server.xml:222]:
```suggestion
     follow the procedure below.
```

File: xml/task-configure-syslog-server.xml (hunk 37 above)
Comment (dmpop) [reply xml/task-configure-syslog-server.xml:291]: You don't perform steps. You perform tasks and follow steps.

File: xml/task-configure-syslog-server.xml [hunk 47]
```diff
@@ -0,0 +1,364 @@
...
+   the following steps to analyze the problem.
+  </para>
+  <variablelist>
+   <varlistentry>
+    <term>Is <systemitem class="daemon">rsyslog</systemitem> running?</term>
+    <listitem>
+     <para>
+      If you made an error in the configuration of <systemitem
+       class="daemon">rsyslog</systemitem>, the daemon might refuse to start.
+      Check it is running with
```
Comment (dmpop) [inline xml/task-configure-syslog-server.xml:301]:
```suggestion
      Check whether it is running with
```

File: xml/task-configure-syslog-server.xml [hunk 48]
```diff
@@ -0,0 +1,364 @@
...
+     </para>
+    </listitem>
+   </varlistentry>
+  </variablelist>
+ </section>
+
+ <section xml:id="next-configure-central-syslog-server">
+  <title>Next steps</title>
+  <para>
+   This basic setup does not include encryption and is only suitable for
```
Comment (dmpop) [inline xml/task-configure-syslog-server.xml:326]:
```suggestion
   This basic setup does not include encryption and it is only suitable for
```

File: xml/task-configure-syslog-server.xml [hunk 49]
```diff
@@ -0,0 +1,364 @@
...
+   trusted internal networks. TLS encryption is strongly recommended, but
+   requires a certificate infrastructure.
```
Comment (dmpop) [inline xml/task-configure-syslog-server.xml:328]:
```suggestion
   trusted internal networks. TLS encryption is strongly recommended. However this
   requires a certificate infrastructure.
```

File: xml/task-configure-syslog-server.xml [hunk 50]
```diff
@@ -0,0 +1,364 @@
...
+  </para>
+  <para>
+   In this configuration, all messages from remote hosts will be treated the
```
Comment (dmpop) [inline xml/task-configure-syslog-server.xml:331]:
```suggestion
   In this configuration, all messages from remote hosts are treated the
```

File: xml/task-configure-syslog-server.xml (hunk 48 above)
Comment (sknorr) [reply xml/task-configure-syslog-server.xml:326]: In this case, either version works.

File: xml/task-configure-syslog-server.xml (hunk 37 above)
Comment (sknorr) [reply xml/task-configure-syslog-server.xml:291]: My tendency is toward "either version works" here, too.

**Comment by cwickert** [comment]
@dmpop Ignore my review request, I didn't see your comments. Thanks, I'll deal with them tomorrow.
